types/stylelint/index.d.ts

```diff
@@ -741,6 +745,17 @@ declare namespace stylelint {
 		source: string;
 	};
 
+	export type EditInfo = {
+		/**
+		 * The start and end offset of the edit. This is the range that is expected to be sliced out.
+		 */
+		range: [number, number];
```

Comment:
question

Can the end offset and start offset be equal?
i.e. `[1, 1]`

Reply:
See:
- https://github.com/stylelint/stylelint/pull/8234#discussion_r1904713767
- https://github.com/stylelint/stylelint/blob/1b17641fe9d80efc3fa7d7aff36d2646be2bd5ca/lib/utils/rangesOverlap.mjs#L3
- https://github.com/stylelint/stylelint/blob/1b17641fe9d80efc3fa7d7aff36d2646be2bd5ca/lib/utils/narrowFixRange.mjs#L53-L62

In practice I don't see how a range could be a point with the current implementation.
This can only happen, in theory, when inserting text into a node that started out with a point range. As we can't increase the width of a range for such a node.

Nothing will or should ever produce AST nodes from empty strings as that doesn't make sense :)

Reply:
I see.

I was asking because ESLint does something peculiar when `[0]` === `[1]`.
i.e. [it inserts](https://eslint.org/docs/latest/integrate/nodejs-api#-editinfo-type)

Reply:
Yup that is correct, but we have a slightly different implementation.
This difference in implementation is unnoticeable for both rule and tool authors.

For insertions we widen the range so that ranges are always unambiguous, as I have explained above :)